pkg/controllers/management/clusterprovisioner/provisioner.go

```diff
@@ -1037,7 +1039,16 @@ func k3sBasedClusterConfig(cluster *v3.Cluster, nodes []*v3.Node) error {
 			cluster.Spec.K3sConfig.SetStrategy(1, 1)
 		}
 	} else if strings.Contains(cluster.Status.Version.String(), "rke2") {
-		cluster.Status.Driver = apimgmtv3.ClusterDriverRke2
+
+		_, err := p.DaemonsetLister.Get("cattle-system", "rancher")
+		if apierrors.IsNotFound(err) {
+			cluster.Status.Driver = apimgmtv3.ClusterDriverRke2
+		} else if err != nil {
+			return err
+		} else {
+			cluster.Status.Driver = apimgmtv3.ClusterDriverRancherD
+			return nil
```

Comment:
do you need to return here? You will still need to set `cluster.Spec.Rke2Config` if it is rancherd, correct?

Reply:
noo, the rke2-config only contains upgrade strategy, and we need to prevent upgrades so i think there should not be a config assigned to this cluster. 
https://github.com/rancher/rancher/blob/3ccd4d9164bd09640e1cd6ad059008e2a06f778d/pkg/api/norman/customization/cluster/validator.go#L231 
specially with this starting upgrades if the rke2Config exists